apps/io-services-cms-webapp/src/synchronizer/request-sync-cms-handler.ts

```diff
@@ -53,34 +54,36 @@ export const handleQueueItem = (
     parseIncomingMessage,
     E.mapLeft((_) => new Error("Error while parsing incoming message")), // TODO: map as _permanent_ error
     TE.fromEither,
-    TE.chain((item) => {
-      switch (item.kind) {
-        case "LifecycleItemType":
-          return pipe(
-            toServiceLifecycle(item.fsm.state, item),
-            (serviceLifecycle) =>
-              fsmLifecycleClient.override(
-                serviceLifecycle.id,
-                serviceLifecycle
-              ),
-            TE.map((_) => void 0)
-          );
-        case "PublicationItemType":
-          return pipe(
-            toServicePublication(item.fsm.state, item),
-            (servicePublication) =>
-              fsmPublicationClient.override(
-                servicePublication.id,
-                servicePublication
-              ),
-            TE.map((_) => void 0)
-          );
-        default:
-          // eslint-disable-next-line @typescript-eslint/no-unused-vars
-          const _: never = item;
-          throw new Error(`should not have executed this with ${item}`);
-      }
-    }),
+    TE.chainW(
+      RA.traverse(TE.ApplicativePar)((item) => {
```

Comment:
**issue:** if run in parallel mode, it is possible that the FSM publication will be updated (perhaps even as the first operation) and the FSM lifecycle update will fail. This leads to an inconsistent state (e.g.: a published service, but not existing in the FSM Lifecycle), so we should run the two operations in sequence!
```suggestion
      RA.traverse(TE.ApplicativeSeq)((item) => {
```
NB: I'm not sure if the above suggestion can fix the "inconsistent state" problem!

Reply:
74636c0